Resources/Public/JavaScript/paste-reference.js

```diff
@@ -0,0 +1,225 @@
+        }
+      },
+      {
+        text: TYPO3.lang['tx_paste_reference_js.modal.button.pastereference'] || 'Paste as reference',
+        btnClass: 'text-white btn-' + top.TYPO3.Severity.getCssClass(severity),
+        trigger: function (evt, modal) {
+          modal.hideModal();
+          DragDrop.default.onDrop(top.itemOnClipboardUid, $element, evt, 'reference');
+        }
+      }
+    ];
+    if (top.pasteReferenceAllowed * 1 !== 1) {
+      buttons.pop();
+    }
+  } else {
+    content = TYPO3.lang['paste.modal.paste'] || 'Do you want to move the record to this position?';
+    buttons = [
+      {
+        text: TYPO3.lang['paste.modal.button.cancel'] || 'Cancel',
+        active: true,
+        btnClass: 'btn-default',
+        trigger: (evt, modal) => modal.hideModal(),
+      },
+      {
+        text: TYPO3.lang['paste.modal.button.paste'] || 'Move',
+        btnClass: 'btn-' + Severity.getCssClass(severity),
+        trigger: function (evt, modal) {
+          modal.hideModal();
+          DragDrop.default.onDrop(top.itemOnClipboardUid, $element, null);
+        }
+      }
+    ];
+  }
+  if (url !== null) {
+    const separator = (url.indexOf('?') > -1) ? '&' : '?';
+    const params = $.param({data: $element.data()});
+    Modal.loadUrl(title, severity, buttons, url + separator + params);
+  } else {
+    Modal.show(title, content, severity, buttons);
+  }
+};
+
+/**
+ * activates the paste into / paste after and fetch copy from another page icons outside of the context menus
+ */
+Paste.activatePasteIcons = function () {
+
+  $('.t3js-page-new-ce').each(function () { // t3-page-ce-wrapper-new-ce
+
+    if (!$(this).find('.icon-actions-plus').length) { // icon-actions-add
+      return true;
+    }
+
+    // @todo: check what does it do in detail and why?
+    // this does not work, clases have the form ".t3js-page-lang-column-0" - ".t3js-page-lang-column-x"
+    $('.t3js-page-lang-column .t3-page-ce > .t3-page-ce').removeClass('t3js-page-ce');
+
+    if (top.copyFromAnotherPageLinkTemplate) {
+
+      // sorting of the buttons is important, else the modal for the first one is not working correctly
+      // OLD: $(this).append(top.copyFromAnotherPageLinkTemplate);
+      if ($(this).find('button.t3js-paste').length) {
+        $(this).find('button.t3js-paste').after(top.copyFromAnotherPageLinkTemplate);
+      } else {
+        $(this).append(top.copyFromAnotherPageLinkTemplate);
+      }
+
+      // Add modal, functionality of the modal itself is not done here,
+      // but rather in paste-reference-drag-drop and triggered by
+      // the custom EventListener 'message' (see downwards)
+      if ($(this).find('button.t3js-paste-new').length) {
+        $(this).find('button.t3js-paste-new').on('click', function (evt) {
+          evt.preventDefault();
+          onReady.copyFromAnotherPage($(this));
+        });
+      }
+    }
+
+    /*
+    // disable default click-EventListener
```

Comment:
Not removed outcommented code. (code smell).

Reply:
Here I removed the wrapping comment to activate the code again: d64e18988